rewatch/src/build/packages.rs

```diff
@@ -237,43 +239,27 @@ fn get_source_dirs(source: config::Source, sub_path: Option<PathBuf>) -> AHashSe
     source_folders
 }
 
-pub fn read_config(package_dir: &Path) -> Result<config::Config> {
+pub fn read_config(package_dir: &Path) -> Result<(Config, PathBuf)> {
     let rescript_json_path = package_dir.join("rescript.json");
     let bsconfig_json_path = package_dir.join("bsconfig.json");
 
     if Path::new(&rescript_json_path).exists() {
-        config::Config::new(&rescript_json_path)
+        Config::new(&rescript_json_path).map(|c| (c, rescript_json_path))
```

Comment:
This feels a bit weird. Why not have the original path as a property on the `Config` you can just take from there? 

Or even put this logic inside the `new` function on `Config`.